flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -180,7 +177,8 @@ public void registerTable(String name, Table table) {
 		}
 
 		CatalogBaseTable tableTable = new QueryOperationCatalogView(table.getQueryOperation());
-		registerTableInternal(name, tableTable);
+		CatalogManagerCommands.createTable(tableTable, false)
+			.executeIn(catalogManager, getTemporaryTablePath(name));
```

Comment:
Why can't we integrate `CatalogManagerCommands` into `CatalogManager`? The first argument of `executeIn` is always the same and basically implements a member that should be part of the manager itself.

Reply:
Good idea